material-stepper/src/main/java/com/stepstone/stepper/type/TabsStepperType.java

```diff
@@ -57,11 +57,11 @@ public void onStepSelected(int newStepPosition) {
      * {@inheritDoc}
      */
     @Override
-    public void onNewAdapter(@NonNull AbstractStepAdapter<?> stepAdapter) {
+    public void onNewAdapter(@NonNull StepAdapter<?> stepAdapter) {
         List<Integer> titles = new ArrayList<>();
         final int stepCount = stepAdapter.getCount();
         for (int i = 0; i < stepCount; i++) {
-            final Step step = stepAdapter.getItem(i);
+            final Step step = (Step) stepAdapter.createStep(i);
```

Comment:
I've given some thought about it, maybe throwing an error on getItem because of invalid cast ?